### PR TITLE
Symintified all functions, not including factory functions

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #86145
* #86139
* __->__ #86078
* #86094
* #86114

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by Chillee** [comment]
Well... it passes tests?

**Review by Chillee — APPROVED** [review]
Surprisingly, it looks like it passes tests? Approving so it can run the land-time tests.

File: .github/ci_commit_pins/xla.txt [hunk 1]
```diff
@@ -1 +1 @@
-96c96ce89d2731846b733054f3cef8915f31a609
+f2b36df6a1a80137eff7644e6d0f4eeb7ff429d6
```
Comment (albanD) [inline .github/ci_commit_pins/xla.txt:1]: Is the xla update necessary here?

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 2]
```diff
@@ -142,6 +142,18 @@ int64_t _safe_size(IntArrayRef sizes, IntArrayRef dim) {
   return size;
 }
 
+c10::SymInt _safe_size(c10::SymIntArrayRef sizes, c10::IntArrayRef dim) {
+  c10::SymInt size = 1;
+  if (sizes.size() == 0) {
+    return 1;
```
Comment (albanD) [inline torch/csrc/autograd/FunctionsManual.cpp:148]: nit return size;

**Review by albanD — APPROVED** [review]
SGTM

File: .github/ci_commit_pins/xla.txt (hunk 1 above)
Comment (ezyang) [reply .github/ci_commit_pins/xla.txt:1]: yes, see https://github.com/pytorch/xla/pull/4067

File: torch/csrc/autograd/FunctionsManual.cpp (hunk 2 above)
Comment (ezyang) [reply torch/csrc/autograd/FunctionsManual.cpp:148]: fwiw this is symmetric with the original

**Pull request closed without merging**